src/creeps/Zombie.cpp

```diff
@@ -170,6 +218,113 @@ void Zombie::generateMove() {
+                //cycle throught the walking images
+                if (getSrcRect().x < ZOMBIE_NEXT_STEP) {
+                    setSrcRect(getSrcRect().x + ZOMBIE_WIDTH, getSrcRect().y, ZOMBIE_WIDTH, ZOMBIE_HEIGHT);
+                } else {
+                    setSrcRect(ZOMBIE_WIDTH, getSrcRect().y, ZOMBIE_WIDTH, ZOMBIE_HEIGHT);
+                }
+            }
+            break;
+        case ZombieDirection::DIR_RD:
+            if (getSrcRect().x == ZOMBIE_FRONT_RIGHT) {
+                setSrcRect(ZOMBIE_WIDTH, getSrcRect().y, ZOMBIE_WIDTH, ZOMBIE_HEIGHT);
+            } else if (frameCountZombie % FRAME_COUNT_ZOMBIE == 0) {
+                if (getSrcRect().x < ZOMBIE_NEXT_STEP) {
+                    setSrcRect(getSrcRect().x + ZOMBIE_WIDTH, getSrcRect().y, ZOMBIE_WIDTH, ZOMBIE_HEIGHT);
+                } else {
+                    setSrcRect(ZOMBIE_WIDTH, getSrcRect().y, ZOMBIE_WIDTH, ZOMBIE_HEIGHT);
+                }
+            }
+            break;
+        case ZombieDirection::DIR_D:
+            if (getSrcRect().x == ZOMBIE_FRONT) {
+                setSrcRect(ZOMBIE_WIDTH, getSrcRect().y, ZOMBIE_WIDTH, ZOMBIE_HEIGHT);
+            } else if (frameCountZombie % FRAME_COUNT_ZOMBIE == 0) {
+                if (getSrcRect().x < ZOMBIE_NEXT_STEP) {
+                    setSrcRect(getSrcRect().x + ZOMBIE_WIDTH, getSrcRect().y, ZOMBIE_WIDTH, ZOMBIE_HEIGHT);
+                } else {
+                    setSrcRect(ZOMBIE_WIDTH, getSrcRect().y, ZOMBIE_WIDTH, ZOMBIE_HEIGHT);
+                }
+            }
+            break;
+        case ZombieDirection::DIR_LD:
+            if (getSrcRect().x == ZOMBIE_FRONT_LEFT) {
+                setSrcRect(ZOMBIE_WIDTH, getSrcRect().y, ZOMBIE_WIDTH, ZOMBIE_HEIGHT);
+            } else if (frameCountZombie % FRAME_COUNT_ZOMBIE == 0) {
+                if (getSrcRect().x < ZOMBIE_NEXT_STEP) {
+                    setSrcRect(getSrcRect().x + ZOMBIE_WIDTH, getSrcRect().y, ZOMBIE_WIDTH, ZOMBIE_HEIGHT);
+                } else {
+                    setSrcRect(ZOMBIE_WIDTH, getSrcRect().y, ZOMBIE_WIDTH, ZOMBIE_HEIGHT);
+                }
+            }
+            break;
+        case ZombieDirection::DIR_L:
+            if (getSrcRect().x == ZOMBIE_LEFT) {
+                setSrcRect(ZOMBIE_WIDTH, getSrcRect().y, ZOMBIE_WIDTH, ZOMBIE_HEIGHT);
+            } else if (frameCountZombie % FRAME_COUNT_ZOMBIE == 0) {
+                if (getSrcRect().x < ZOMBIE_NEXT_STEP) {
+                    setSrcRect(getSrcRect().x + ZOMBIE_WIDTH, getSrcRect().y, ZOMBIE_WIDTH, ZOMBIE_HEIGHT);
+                } else {
+                    setSrcRect(ZOMBIE_WIDTH, getSrcRect().y, ZOMBIE_WIDTH, ZOMBIE_HEIGHT);
+                }
+            }
+            break;
+        case ZombieDirection::DIR_LU:
+            if (getSrcRect().x == ZOMBIE_BACK_LEFT) {
+                setSrcRect(ZOMBIE_WIDTH, getSrcRect().y, ZOMBIE_WIDTH, ZOMBIE_HEIGHT);
+            } else if (frameCountZombie % FRAME_COUNT_ZOMBIE == 0) {
```

Comment:
Again, these cases only have a single if condition making them different
See if you can pull it out of the switch statement using a loop or something
Because this is a lot for code duplication
Hell, make a wrapper for it and pass in the direction so that you can perform your check properly
There's no need to copy like 10 lines over and  over like this